src/utils.c

```diff
@@ -97,6 +97,32 @@ set_fd_nonblocking(int fd)
 }
 
 /*!
+ * Store short integer as big endian in buffer
+ *
+ * \param buf Buffer to store the value in
+ * \param val Short Integer to be converted to big endian
+ */
+void
+set_big_endian_16(uint8_t *buf, uint16_t val)
```

Comment:
I'd consider simply returning if buf is `NULL` (or making the function return a `bool`).